### PR TITLE
Normalize trytes when decoding trytes to ascii

### PR DESCRIPTION
**Pull request by Roconda**
Found out the hard way that tryte string length should be even, wasn't able to decode the tag and message.
This method appends a '9' to trytes to make the string length even.

### CODE REVIEW

**Comment by Roconda** [comment]
Fixes #96

File: lib/utils/asciiToTrytes.js [hunk 1]
```diff
@@ -93,7 +93,21 @@ function fromTrytes(inputTrytes) {
...
 
+/**
+ * This method normalizes trytes by appending '9'
+ * in case a tryte string length is odd.
+ *
+ * @param {string} inputTrytes
+ * @returns {string} A tryte with an even length.
+ */
+function normalizeTrytes(inputTrytes) {
+    if ( inputTrytes.length % 2 ) return normalizeTrytes(inputTrytes += '9')
```
Comment (th0br0) [inline lib/utils/asciiToTrytes.js]: This recursion is unneccessary.

File: lib/utils/asciiToTrytes.js [hunk 2]
```diff
@@ -71,7 +71,7 @@ function fromTrytes(inputTrytes) {
     if ( typeof inputTrytes !== 'string' ) return null
 
     // If input length is odd, return null
-    if ( inputTrytes.length % 2 ) return null
+    if ( inputTrytes.length % 2 ) inputTrytes = normalizeTrytes(inputTrytes)
```
Comment (th0br0) [inline lib/utils/asciiToTrytes.js]: I don't think we need to outsource this into a separate method.
Also, 'normalize' is a misleading label - you're not normalizing the trytes themselves.

**Comment by laubsauger** [comment]
This one is really nasty, i've been stumbling over it today. 
@Roconda Are you going to look into this again now that there is PR-Feedback?

**Comment by Roconda** [comment]
Hi @laubsauger, kinda forgot about this PR, thanks for reminding.

It's true null chars are appended(because of the 9's). It really depends on the payload if you want them append or not. Most cases people probably use JSON where it's not desired to have those appended. Maybe @th0br0 has an idea depending on the payload(maybe check if its JSON parsable).

Maybe something like `iota.utils.fromTrytes('.....').replace(/\0[\s\S]*$/g,'')`, a bit safer than your current work around. Will have some time in the evening this week to take a look.

**Comment by obany** [comment]
Closing as these changes are on an unmaintained branch of the library, which will be incompatible with the new Chrysalis release of the network.

**Pull request closed without merging**